balancer/rls/cache_test.go

```diff
@@ -242,3 +242,50 @@ func (s) TestDataCache_ResetBackoffState(t *testing.T) {
 		t.Fatalf("unexpected diff in backoffState for cache entry after dataCache.resetBackoffState(): %s", diff)
 	}
 }
+
+var cacheEntriesMetricsTests = []*cacheEntry{
+	{size: 3},
+	{size: 3},
+	{size: 3},
+	{size: 3},
+	{size: 3},
+}
+
+func (s) TestDataCache_Metrics(t *testing.T) {
+	tmr := stats.NewTestMetricsRecorder(t)
+	dc := newDataCache(50, nil, tmr, "")
+
+	dc.updateRLSServerTarget("rls-server-target")
+	for i, k := range cacheKeys {
+		dc.addEntry(k, cacheEntriesMetricsTests[i])
+	}
+
+	// 5 total entries, size 3 each, so should record 5 entries and 15
+	// size.
+	tmr.AssertDataForMetric("grpc.lb.rls.cache_entries", 5)
+	tmr.AssertDataForMetric("grpc.lb.rls.cache_size", 15)
+
+	// Resize down the cache to 3 entries. This should scale down the cache to 3
+	// entries with 3 bytes each, so should record 3 entries and 9 size.
+	dc.resize(9)
+	tmr.AssertDataForMetric("grpc.lb.rls.cache_entries", 3)
+	tmr.AssertDataForMetric("grpc.lb.rls.cache_size", 9)
```

Comment:
This relies on having the same size for every entry? Or does order matter?

Reply:
Yeah it relies. I think operations are deterministic if I tweak the numbers, but I don't know if we'll have that guarantee in the future. Let me play around with this some.